lib/pluginmanager/install_strategy/elastic_xpack.rb

```diff
@@ -0,0 +1,41 @@
+# encoding: utf-8
+# In the context of the plugin manager no dependencies are currently loaded.
+require_relative "../../../logstash-core/lib/logstash/version"
+require "pluginmanager/xpack/remote_installer"
+require "pluginmanager/utils/http_client"
+require "pluginmanager/ui"
+require "net/http"
+require "uri"
+
+module LogStash module PluginManager module InstallStrategy
+  class ElasticXpack
+    ELASTIC_XPACK_BASE_URI = ENV["LOGSTASH_PACK_URL"] || "https://artifacts.elastic.co/downloads/logstash"
+    XPACK_EXTENSION = "zip"
+
+    class << self
+      def pack_uri(plugin_name)
+        url = "#{ELASTIC_XPACK_BASE_URI}/#{plugin_name}-#{LOGSTASH_VERSION}.#{XPACK_EXTENSION}"
+        URI.parse(url)
+      end
+
+      def match?(plugin_name)
+        uri = pack_uri(plugin_name)
+
+        PluginManager.ui.debug("Looking if package named: #{plugin_name} exists at #{uri}")
+
+        if Utils::HttpClient.remote_file_exist?(uri)
+          PluginManager.ui.debug("Found package at: #{uri}")
+          LogStash::PluginManager::Xpack::RemoteInstaller.new(uri)
+        else
+          PluginManager.ui.debug("Package not found at: #{uri}")
+        end
+      rescue SocketError => e
+        # This probably means there is a firewall in place of the proxy is not correctly configured.
+        # So lets skip this strategy but log a meaningful errors.
+        PluginManager.ui.debug("SocketError, skipping Elastic Xpack, exception: #{e}")
```

Comment:
We don't log whole exception object, since we have string interpolation at hand it will call e.message in this context